src/crystal/system/process.cr

```diff
@@ -0,0 +1,21 @@
+module Crystal
+  # :nodoc:
+  module System
+    # :nodoc:
+    module Process
+      # Returns possible location of executable path
+      # def self.executable_path_impl
+    end
+  end
+end
+
+{% if flag?(:darwin) %}
+  require "./unix/darwin_process"
+{% elsif flag?(:freebsd) %}
+  require "./unix/freebsd_process"
+{% elsif flag?(:linux) %}
+  require "./unix/linux_process"
+{% else %}
+  # TODO: restrict on flag?(:unix) after crystal > 0.22.0 is released
```

Comment:
Actually the TODO should be removed and the `else` branch implement or require the generic "find executable in PATH" solution, which is valid for both UNIX and Windows for example.